### ISSUE TITLE
Rebalance

### ISSUE THREAD

**Issue by rkurbatov** [ciplogic/fheroes2enh#22]
I accidentally provided this PR into the main repo instead of the local one and it was merged: https://github.com/ciplogic/fheroes2enh/pull/21

It has pretty strong game rebalance that I'd wanted to be discussed.

The main source for the PR is this patch: https://www.celestialheavens.com/forum/7/8724 with some my additions.

**MAIN IDEAS**

MIGHT fractions are weak and not usable. The typical scenario is to get a warlock hero as the main spellcaster, barbarian as a scout, sorceress as a naval scout and ignore knight if possible. Necromancers are good to get a huge pack of skeletons for the long game. This has to be changed: knights and barbarians cannot become good casters, but they can have good crude force abilities lost by MAGIC fractions. Necromancers and sorcerers should be more versatile.

Unit stats are changed a little bit. Almost all knight and some barbarian units are buffed, warlock and wizard units are nerfed with 6-tier unit amount decreased. Now they simply don't replenish without well and become more valuable in terms of loss. We can make well as a requirement for 6 tier or change well behavior so it increases the population of tiers 1-5 only.

**PRIMARY SKILLS**

Primary skills are mostly the same. Initial skill points are changed for wizards (0/0/2/3 instead of 0/1/2/2) and sorcerers (0/1/2/2 instead of 0/0/2/3). Wizards need more magic points from the very beginning while sorcerers rely more on defense.

Level 10+ points are distributed more evenly because magic becomes more important on later stages:

|          | L1 percentage |   L10+ percentage |
|----------|---------------|-------------------|
| Knight  | 35/45/10/10   |  25/25/25/25      |
| Barbarian | 55/35/5/5   | 25/25/25/25 (was 30/30/20/20) |
| Sorceress | 20/15/32/33 (was 10/10/30/50) | 25/20/25/30 (was 20/20/30/30) |
| Necromancer | 20/20/30/30 (was 15/15/35/35) | 25/25/25/25 |
| Warlock | 15/15/40/30 (was 10/10/50/30)| 20/20/30/30 |
| Wizard | 15/15/35/35 (was 10/10/40/40) | 20/20/30/30 | 

Secondary skills are changed more. There are 14 secondary skills with Necromancy not available to all except for Necromancers and Warlocks and Leadership not available to Necromancers only thus every fraction can select among 13 skills and warlocks - among 14. Now every fraction has 2 skills not provided during skill points selection with experience growth.
That's also uneven because these skills can be obtained from witch hut (except for Necromancy). A simple solution is to enable Necromancy so that it can be learned by any fraction. Another approach is to generate in a hut a new skill among available for every new hero to support diversity :)

*Knights* now cannot learn Necromancy and Mysticism. Initial skills are same - Ballistics and Leadership. Archery probability increased.
*Barbarians* now cannot learn Necromancy and Eagle eye. Initial skills changed to Pathfinding and Scouting instead of advanced Pathfinding. Estates probability decreased, Pathfinding - increased.
*Sorceress* now cannot learn Necromancy and Leadership. Initial skills changed to Wisdom and Luck instead of basic Wisdom and advanced Navigation. Navigation was completely useless on the maps without sea while sorceress should become more might and luck oriented. Ballistics and Navigation probability decreased, Luck probability increased.
*Necromancer* now cannot learn Leadership and Scouting. Initial skills are same - Wisdom and Necromancy. Ballistics probability decreased, Diplomacy - increased.
*Warlock* now cannot learn Archery and Luck. Serious nerf. There is only one weak shooter in the army so no need for that. 
Initial skills are basic Wisdom and basic Eagle Eye instead of advanced Wisdom and Scouting. Ballistics (much) and Scouting probability is decreased, Eagle Eye, Leadership and Estates increased.
*Wizard* now cannot learn Necromancy and Ballistics. They have lots of shooters and can rely on magic (Earthquake). Archery and Estates probability is increased.

Further changes are possible with even more biased secondary skills points distribution to make fraction heroes less universal and allow different strategies.

Also, stats for some skills are changed to make them more valuable.

*Scouting* now has 3 cells radius on an advanced level (was 2) and 5 on expert (was 3) just like in Heroes III.
*Eagle eye* now has 25%/40%/65% probability to recognize spell rather instead of 20/30/40 values.
There are ideas to make some skills even more valuable but they require more code changes, not just numbers.
*Necromancy* algorithm should be changed and take into account not only number of killed units but their stats to
prevent huge necro army gross after fights with peasants and other tear-1.
*Ballistics* and *Archery* can allow the hero to manage ballista and turrets just like in Heroes 3.
*Diplomacy* should be rewritten, the current price of joinable units is a way too high.

Now unit and building changes

**KNIGHTS**

Almost all units are buffed with little cost increase. High tear buildings are less expensive.

*Peasant* growth increased 12 => 26 (totally 36 with well and farm), gold cost decreased 20 => 15, speed increased: *very slow => slow*. If that's not enough I'd added some special ability to this unit to make it valuable. Halflings and centaurs are shooters, sprites are flying and enemies don't retaliate, goblins are strong, skeletons are used en masse and only peasants are useless. Any ideas?
 
*Archer/Ranger* hitpoints increase: 10 => 11
 
*Pikemen/Veteran Pikemen* hit points, damage and cost increase: 15/20 => 16/21, 3-4 => 3-5, 200/250 => 210/260
 
*Swordsmen/Master swordsmen* hit points, damage and cost increase: 25/30 => 26/31, 4-6 => 4-7, 250/300 => 260/310
 
*Cavalry/Champion* hit points and cost increase: 30/40 => 35/45, 300/375 => 340/415. Heroes 3 improvements with attack bonus based on path champion rode is also possible.
 
*Paladin/Crusader* hit points increase: 50/65 => 60/75 with Paladin cost increase 600 => 675 (Crusader is still 1000).
 
*Cathedral* wood cost decrease - 5000/20/20 => 5000/**10**/20
 
*Upgraded Cathedral* cost decrease from 5000 gold, 10 wood, 10 crystal to 3500 gold, 10 wood, 5 crystal.
 
*Fortifications* cost decrease from 1500 gold, 5 wood, 15 ore to 1500 gold, 0 wood, 10 ore.
 
Knights should be more mighty units that rely on Ballistics, Archery, and Leadership, having fortified castles.
 
 
**BARBARIANS**

Except for Ogres units are stronger or/and cheaper.

*Goblin* growth is increased from 10 to 12 (totally 22 with well and garbage)

*Orc* hitpoints increase 10 => 11

*Wolf* defence increase 2 => 3 with increased cost 200 => 210

*Ogre Lords* are weaker (60 HP => 55 HP) and cheaper (500 => 475) - this tier 4 unit was too powerful

*Trolls/War Trolls* are cheaper: 600/700 => 575/650

*Cyclops* are stronger: 80HP => 85HP while

*Pyramid* is cheaper: 6000 gold, 20 ore, 20 crystal => 6000 gold, **10** ore, 20 crystal.

*Coliseum* is also cheaper: 2000 gold, 10 ore, 10 wood => 1500 gold, 5 ore, 5 wood

**SORCERESS**

Nature castle units are tougher and powerful mostly. Shooters are always a target.

*Sprite* growth is increased 8 => 10 (totally 20 with well and garden)

*Dwarfs/Battle Dwarfs* are cheaper: 200/250 => 190/235

*Elfs/Grand Elfs* are more powerful: 4/5 attack => 5/7 attack and stronger (15HP => 16HP)

*Greater Druids* has more HP (25 => 26) while

*Upgraded Stonehenge* is more expensive 1500 gold, 5 ore => **2000** gold, 5 ore

*Unicorns* are tougher (40 HP => 45 HP) while more expensive (500 => 515)

*Rainbow* is changed in the price: 1500 gold, 10 crystals to 1500 gold, **5** mercury, **5** crystals

If more buff is required, Archery can become initial skill instead of Luck

**WARLOCKS**

Warlocks are much weaker now.

*Cave* now requires *5* ore additionally to 500 gold.

*Gargoyle* defence is decreased (7 => 6) though

*Crypt* is less expensive now: 1000 gold, 10 ore => 1000 gold, *5* ore.

*Griffin* is more expensive: 300 => 320

*Hydra* is less expensive: 800 => 775

*Green/Red/Black Dragons* have basic growth set to 0, totally 2 in case well is built. 1 less dragon per week.
Additionally hit points decreased. Red dragon: 250 => 240, Black dragon: 300 => 280.

*Green tower* is less expensive: 15 000 gold, 30 ore, 20 sulfur => **13 000** gold, 30 ore, 20 sulfur while

*Red/Black tower* upgrade is more expensive: 5000 gold, 5 ore, 10 sulfur => **6000** gold, 5 ore, 10 sulfur. Total black
 tower price is the same, but green dragons are more usable now with red/black nerf.
 
*Dungeon* cost is much increased. Additionally to 3000 gold, 5 wood and 10 ore you should pay 3 units of each rare resource.

On the other hand, stupid Scouting is removed and Eagle Eye is more useful (instead of Wizards library).

**WIZARDS**

*Halflings* cost increased: 50 => 55.
*Boar* damage increase: 2-3 => 2-4. Blessing is as also good on them now.
*Roc* defence is increased: 7 => 8 
*Upgraded Ivory Tower* is less expensive: 4000 gold, 5 wood, 5 ore => *3000* gold, 5 wood, 5 ore

Most changes are about *Giants/Titans*. Their basic growth is set to 0, so only 2 of them are provided by well. 1 less big guy per week.

*Giants* are more usable with HP growth: 150 => 155 and less expensive *Cloud Castle*: 12 500 gold => **7 500** gold, while same 5 ore, 5 wood and 20 gems.

*Titans* are less tough now: 300 HP => 280 HP while less expensive: 5000 gold, 2 gems => 4500 gold, 2 gems. But

*Upgraded Cloud Castle* is hell pricy: 12 500 gold => *16 500* gold additionally to 5 ore, 5 wood, 20 gems.

All of this makes wizard a little bit stronger on the early game while little bit weaker on late game.

**NECROMANCERS**

Changes here are most discussable.

*Skeletons* are cheaper: 75 => 70 that allows buying more in light of possible Necromancy nerf instead of raising them.

*Zombies* are faster: Very slow => Slow

Both *Zombie/Mutant Zombie* defence increased: 2 => 3 while price decreased: 150/200 => 140/190.

*Mummy/Royal Mummy* defence increased: 6 => 7

*Upgraded Mansion* is cheaper: 4000 gold, 5 wood, 10 crystals, 10 gems => *3500* gold, 5 wood, **10** mercury, 5 crystals.

*Lich* hit points increased 25 => 30 with defence decreased 12 => 11

Both *Lich/Power Lich* are less expensive 750/900 => 650/750 and

*Upgraded Mausoleum* is too: 3000 gold, 5 ore, 5 crystal => **2500** gold, 5 ore, **5** sulfur.

*Bone Dragon* nerfed with both HP: 150 => 140 and higher price: 1500 => 1600.

*Laboratory* is also more expensive: 10 000 gold, 5 mercury => **12 000** gold, **10** mercury.

**COMMON BUILDINGS**

Mage guild upgrade price is changed:

Lvl 2 - 1000/5/5/4 rare => 1500/5/5/2 rare - more gold, less rare resources.
Lvl 3 - 1000/5/5/6 rare => 1500/5/5/4 rare 
Lvl 4 - 1000/5/5/8 rare => 2000/5/5/6 rare
Lvl 5 - 1000/5/5/10 rare => 2000/5/5/8 rare

Shipyard is less wood-dependent: 2000 gold, 20 wood => 2000 gold, *15* wood.

**SPELLS**

*Fireball/Fireblast/Cold ring* require less SPs 9/15/9 => 8/14/8 and have stronger damage ratio: 10 => 13.

*Chain lightning* is made level 5 spell and requires more SPs: 15 => 200

*Hypnotize* is made level 4 spell instead (discussable)

*Mirror Image* is less expensive: 25 => 20

Other spells made more "expensive" in terms of SP:

*Animate Dead*: 10 => 15

*Resurrect*: 12 => 13

*Resurrect true*: 15 => 20

*Blind*: 6 => 7

*Holy Shout*: 12 => 14

*Elemental Storm*: 15 => 20

*Meteor Shower*: 15 => 18

*Armageddon*: 20 => 30

*Dimension Door*: 10 => 22

Further ways to improve spell system.

Some castles should not have some spells available like Bless/Holy Word/Holy Shout for Necromancers (useless) and Armageddon for Warlocks (too strong).
Due to availability of the standard level 1 spell for every magic fraction (Haste/Curse/Bless/Stoneskin) corresponding castle should either have that level 1 spell in Mage Guild or not have it for every fraction castle. That allows more even spell distribution.

**Comment by ciplogic** [comment]
I hope I will make a new release (over this or next weekend) and I think I should bump the version to 1.1 for this reason. It is an important improvement in gameplay,

**Comment by ciplogic** [comment]
I understood also the PR and I merged it for a critical reason: even it would be unbalanced (which feels like an improvement to me), we can revert it if we get bad feedback, but many items are really felt after people play the game.

In 1.5 years from first release I got just one comment on changes which fixes gameplay, (to have an option to revert to defaults) but no one else did this comment. But purists can play always the original game.
